sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -171,58 +178,74 @@ export class EventHubClient {
    */
   constructor(
     host: string,
-    entityPath: string,
+    eventHubPath: string,
     credentials: ApplicationTokenCredentials | UserTokenCredentials | DeviceTokenCredentials | MSITokenCredentials,
     options?: ClientOptions
   );
   constructor(
-    host: string,
-    entityPath: string,
-    tokenProviderOrCredentials:
+    hostOrConnectionString: string,
+    eventHubPathOrOptions?: string | ClientOptions,
+    tokenProviderOrCredentials?:
       | TokenProvider
       | ApplicationTokenCredentials
       | UserTokenCredentials
       | DeviceTokenCredentials
       | MSITokenCredentials,
     options?: ClientOptions
   ) {
-    if (!host) {
-      throw new Error(
-        "Please provide a fully qualified domain name of your Event Hub instance for the 'host' parameter."
-      );
-    }
-
-    if (!entityPath) {
-      throw new Error("Please provide a value for the 'entityPath' parameter.");
-    }
+    let connectionString;
+    let tokenProvider = {} as TokenProvider;
+    hostOrConnectionString = String(hostOrConnectionString);
 
-    if (!tokenProviderOrCredentials) {
-      throw new Error("Please provide either a token provider or a valid credentials object.");
-    }
+    if (typeof eventHubPathOrOptions === "string") {
```

Comment:
One last suggestion :)

What do you think of swapping the if and the else blocks here?
i.e 
```javascript
if (typeof eventHubPathOrOptions !== "string") {
   // hostOrConnectionString is connection string
} else {
   // hostOrConnectionString is host
}
```

This would match with the order of the constructor overloads we have and make it event more easier to follow the code from top to bottom